### PR TITLE
FIX-#2110: get rid of 'NotImplementedError' at OmniSci query compiler

### LINKED ISSUE #2110: Replace raising `NotImplementedError` methods to calling BaseQueryCompiler methods in OmniSci query compiler
Currently `DFAlgQueryCompiler` raising `NotImplementedError` when obtain parameters, implementation of which is not ready yet. It's needed to replace all this raisings to
```python
super().method(*args, **kwargs)
```

### PR DESCRIPTION
**Pull request by dchigarev**
Signed-off-by: Dmitry Chigarev <dmitry.chigarev@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #2110 
- [x] tests added and passing

### CODE REVIEW

File: modin/experimental/backends/omnisci/query_compiler.py [hunk 1]
```diff
@@ -342,7 +342,7 @@ def dt_month(self):
     def _bin_op(self, other, op_name, **kwargs):
         level = kwargs.get("level", None)
         if level is not None:
-            raise NotImplementedError(f"{op_name} doesn't support levels")
+            getattr(super(), op_name)(other=other, op_name=op_name, **kwargs)
```
Comment (ienkovich) [inline modin/experimental/backends/omnisci/query_compiler.py:345]: `return` is missing here

File: modin/experimental/backends/omnisci/query_compiler.py [hunk 2]
```diff
@@ -424,11 +424,8 @@ def setitem(self, axis, key, value):
         Returns:
              A new QueryCompiler
         """
-        if axis == 1:
-            raise NotImplementedError("setitem doesn't support axis 1")
-
-        if not isinstance(value, type(self)):
-            raise NotImplementedError("unsupported value for setitem")
+        if axis == 1 or not isinstance(value, type(self)):
+            super().setitem(axis=axis, key=key, value=value)
```
Comment (ienkovich) [inline modin/experimental/backends/omnisci/query_compiler.py:428]: `return` is missing here

File: modin/experimental/backends/omnisci/query_compiler.py [hunk 3]
```diff
@@ -443,12 +440,10 @@ def insert(self, loc, column, value):
...
 
         Returns:
-            A new PandasQueryCompiler with new data inserted.
+            A new DFAlgQueryCompiler with new data inserted.
         """
         if is_list_like(value):
-            raise NotImplementedError(
-                "non-scalar values are not supported by insert yet"
-            )
+            super().insert(loc=loc, column=column, value=value)
```
Comment (ienkovich) [inline modin/experimental/backends/omnisci/query_compiler.py:446]: `return` is missing here

**Review by ienkovich — CHANGES_REQUESTED** [review]

File: modin/experimental/backends/omnisci/query_compiler.py (hunk 1 above)
Comment (dchigarev) [reply modin/experimental/backends/omnisci/query_compiler.py]: fixed

File: modin/experimental/engines/omnisci_on_ray/test/test_dataframe.py [hunk 4]
```diff
@@ -398,7 +399,19 @@ def insert(df, **kwargs):
...
+        def applier(df, **kwargs):
+            df.insert(0, "new_int", 10)
+            df.insert(0, "new_float", 5.5)
+            df.insert(0, "new_list_like", np.arange(len(df)))
+            df.insert(0, "qc_column", df["new_int"])
+            return df
+
+        # setting `force_laze=False`, because `to_pandas` is not supported
+        # in lazy frame mode
+        run_and_compare(applier, data=self.data, force_lazy=False)
```
Comment (dchigarev) [inline modin/experimental/engines/omnisci_on_ray/test/test_dataframe.py:414]: @ienkovich Am I right that `default_to_pandas` is not supported in lazy execution mode?
https://github.com/modin-project/modin/blob/f46382c31d2bea7706e6242143817f955e1eac52/modin/experimental/engines/omnisci_on_ray/frame/data.py#L1090-L1095

Comment (ienkovich) [reply modin/experimental/engines/omnisci_on_ray/test/test_dataframe.py:414]: Right, we set 'lazy' mode in tests to make sure computations are done lazily and `default_to_pandas` calls are not lazy.

File: modin/experimental/engines/omnisci_on_ray/test/test_dataframe.py [hunk 5]
```diff
@@ -1209,6 +1227,9 @@ def sort(df, cols, ignore_index, index_cols, ascending, **kwargs):
             ignore_index=ignore_index,
             index_cols=index_cols,
             ascending=ascending,
+            # we're excecting to fallback to pandas in that case,
```
Comment (ienkovich) [inline modin/experimental/engines/omnisci_on_ray/test/test_dataframe.py:1230]: typo here

**Review by ienkovich — DISMISSED** [review]
The patch looks good! Thanks!

File: modin/experimental/engines/omnisci_on_ray/test/test_dataframe.py (hunk 5 above)
Comment (dchigarev) [reply modin/experimental/engines/omnisci_on_ray/test/test_dataframe.py]: fixed

**Review by ienkovich — APPROVED** [review]

**Pull request merged**